tensorflow/python/layers/convolutional.py

```diff
@@ -1200,16 +1201,238 @@ def conv2d_transpose(inputs,
+               bias_regularizer=None,
+               activity_regularizer=None,
+               trainable=True,
+               name=None,
+               **kwargs):
+    super(Conv3DTranspose, self).__init__(
+        rank=3,
+        filters=filters,
+        kernel_size=kernel_size,
+        strides=strides,
+        padding=padding,
+        data_format=data_format,
+        activation=activation,
+        use_bias=use_bias,
+        kernel_initializer=kernel_initializer,
+        bias_initializer=bias_initializer,
+        kernel_regularizer=kernel_regularizer,
+        bias_regularizer=bias_regularizer,
+        activity_regularizer=activity_regularizer,
+        trainable=trainable,
+        name=name, **kwargs)
+
+  def call(self, inputs):
+    inputs_shape = array_ops.shape(inputs)
+    batch_size = inputs_shape[0]
+    if self.data_format == 'channels_first':
+      c_axis, d_axis, h_axis, w_axis = 1, 2, 3, 4
+    else:
+      c_axis, d_axis, h_axis, w_axis = 4, 1, 2, 3
+
+    depth, height, width = inputs_shape[d_axis], inputs_shape[h_axis], \
+        inputs_shape[w_axis]
+    kernel_d, kernel_h, kernel_w = self.kernel_size
+    stride_d, stride_h, stride_w = self.strides
+
+    def get_deconv_dim(dim_size, stride_size, kernel_size, padding):
+      if isinstance(dim_size, ops.Tensor):
+        dim_size = math_ops.multiply(dim_size, stride_size)
+      elif dim_size is not None:
+        dim_size *= stride_size
+
+      if padding == 'valid' and dim_size is not None:
+        dim_size += max(kernel_size - stride_size, 0)
+      return dim_size
+
+    # Infer the dynamic output shape:
+    out_depth = get_deconv_dim(depth, stride_d, kernel_d, self.padding)
+    out_height = get_deconv_dim(height, stride_h, kernel_h, self.padding)
+    out_width = get_deconv_dim(width, stride_w, kernel_w, self.padding)
+
+    if self.data_format == 'channels_first':
+      output_shape = (batch_size, self.filters, out_depth, out_height,
+                      out_width)
+      strides = (1, 1, stride_d, stride_h, stride_w)
+    else:
+      output_shape = (batch_size, out_depth, out_height, out_width,
+                      self.filters)
+      strides = (1, stride_d, stride_h, stride_w, 1)
+
+    output_shape_tensor = array_ops.stack(output_shape)
+    outputs = nn.conv3d_transpose(
+        inputs,
+        self.kernel,
+        output_shape_tensor,
+        strides,
+        padding=self.padding.upper())
+
+    # Infer the static output shape:
+    out_shape = inputs.get_shape().as_list()
+    out_shape[c_axis] = self.filters
+    out_shape[d_axis] = get_deconv_dim(
+        out_shape[d_axis], stride_d, kernel_d, self.padding)
+    out_shape[h_axis] = get_deconv_dim(
+        out_shape[h_axis], stride_h, kernel_h, self.padding)
+    out_shape[w_axis] = get_deconv_dim(
+        out_shape[w_axis], stride_w, kernel_w, self.padding)
+    outputs.set_shape(out_shape)
+
+    if self.bias:
+      outputs = nn.bias_add(
```

Comment:
I believe this would fail in data_format `channels_first` (to be verified). See what we do for `Conv3D` bias.